### PR TITLE
C1 abdelrahman wishlist

### PR DESCRIPTION
**Pull request by abdelrahmanMaged**
@BeshoSmile 
@Reem-Sharkawy 
@Tharwat7 

Please review

### CODE REVIEW

File: project/tager/tager_www/templates/ViewPost.html [hunk 1]
```diff
@@ -131,43 +159,37 @@ <h3 id="myModalLabel">Buyers Interested in Your Post:</h3>
...
 
-</div>
 
 
 </div>
-{% block script%}
+
 
     <script type="text/javascript">
+    var eligible_to_add_to_the_wish_list = {{canwish}}
```
Comment (BeshoyAtef) [inline project/tager/tager_www/templates/ViewPost.html:168]: Please tabbify ur code 

this website will auto do this 
(choose from teh code type JSON and will help yu do it )
http://tools.arantius.com/tabifier

File: project/tager/tager_www/templates/editPost.html [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    type: "POST",
+    data: {
+        "post" : id,
+        "title": title,
+    }, 
+    success: function(result) {
+    alert("Successfully updated")
+    $('#title1').text($('#title1value').val());
+    $('#title1').show();
+    $('#title1value').hide();
```
Comment (BeshoyAtef) [inline project/tager/tager_www/templates/editPost.html:107]: Please tabbify ur code 

this website will auto do this 
(choose from teh code type JSON and will help yu do it )
http://tools.arantius.com/tabifier

File: project/tager/tager_www/models.py [hunk 3]
```diff
@@ -203,10 +203,45 @@ def report_the_post(self, post, report_reason):
...
+    # It checks whether the user has the prevalage to post or not.
+    #the return type is whether true or false. 
+    # if the user is not verfied the method returns false 
+    # else if the user is premium the method return true without checking any further condition. 
+    # but if the user is not premium but verfied then the method check the user's number of active posts if it is below three the method return true else it returns false.
     def can_post(self):
-        return self.is_verfied
-
+        no_of_user_active_posts =  Post.objects.filter(seller = self).exclude(state='sold').exclude(state='Archived').count()
+        print no_of_user_active_posts
```
Comment (BeshoyAtef) [inline project/tager/tager_www/models.py:215]: Delete this print

File: project/tager/tager_www/templates/editPost.html [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    var title = $('#title1value').val();
+    $.ajax({
+    url: "/editposttitle/",
+    type: "POST",
+    data: {
+        "post" : id,
+        "title": title,
+    }, 
+    success: function(result) {
+    alert("Successfully updated")
```
Comment (BeshoyAtef) [inline project/tager/tager_www/templates/editPost.html:104]: please add more reliable alert

File: project/tager/tager_www/models.py [hunk 5]
```diff
@@ -203,10 +203,45 @@ def report_the_post(self, post, report_reason):
...
+
+    #c1 abdelrahman the method takes self, and post id as an arguments it checks whether this combination is in the table if it is in the query set. 
+    # it returns false meaning that the user can not add this post to the wish list, else it returns true.
+    def add_to_wish_list(self, post_id):
+        if WishList.objects.filter(post=post_id , user = self).exists():
+            return "false"
+        else:
+            return "true"
 
+    #The Method Takes 2 arguments(User who clicked intrested,Post Which the user has clicked the button in)
```
Comment (BeshoyAtef) [inline project/tager/tager_www/models.py]: DELETE THIS METHOD

File: project/tager/tager/urls.py [hunk 6]
```diff
@@ -12,10 +12,13 @@
     
     
     url(r'^logout/$', logout, {'next_page':'/login'}),  
-
+    
     url(r'^$', 'tager_www.views.home'),
     url(r'^main$', 'tager_www.views.main'),
+    url(r'^addtomylist$', 'tager_www.views.add_to_wish_list'),
+    # url(r'^intrested/$', 'tager_www.views.intrested', name="intrested"),
```
Comment (BeshoyAtef) [inline project/tager/tager/urls.py]: ??

**Comment by Reem-Sharkawy** [comment]
#185

**Comment by HalaKamal** [comment]
tage me when u finish the diagram  @AbdelrahmanAyad

**Pull request merged**